src/bindings/c/include/openvino/ov_shape.h

```diff
@@ -0,0 +1,157 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for ov_shape C API
+ *
+ * @file ov_shape.h
+ */
+
+#pragma once
+
+#include "ov_common.h"
+
+// Shape
+/**
+ * @defgroup Shape Shape
+ * @ingroup openvino_c
+ * Set of functions representing of Shape, PartialShape, etc.
+ * @{
+ */
+
+/**
+ * @brief Create a rank object
+ * @ingroup Shape
+ * @param min_dimension The lower inclusive limit for the dimension, for static object you should set same value(>=0)
+ * with max_dimension
+ * @param max_dimension The upper inclusive limit for the dimension, for static object you should set same value(>=0)
+ * with min_dimension
+ * @param ov_status_e a status code.
+ */
+OPENVINO_C_API(ov_status_e) ov_rank_create(ov_rank_t** rank, int64_t min_dimension, int64_t max_dimension);
```

Comment:
The same for this file. Why does `shape.h` defines `rank`, `dimension`, `partial_shape` and etc.?

Reply:
It is the related types, we just group them and don't want to split too much. BTW, for C APIs, I think it is a bad idea to split too much.

Reply:
I explained the idea in previous comment.

Reply:
you mean we need split at least 2 header files, one is for shape and another for partial_shape?

Reply:
done